plugins/engine-configuration/impl/src/main/java/org/pentaho/di/engine/configuration/impl/pentaho/scheduler/SchedulerRequest.java

```diff
@@ -129,77 +135,87 @@ private String getFullPath( AbstractMeta meta ) {
 
   StringEntity buildSchedulerRequestEntity( AbstractMeta meta )
           throws UnsupportedEncodingException, UnknownParamException {
+
     String filename = getFullPath( meta );
 
-    StringBuilder sb = new StringBuilder();
-    sb.append( "<jobScheduleRequest>\n" );
-    sb.append( "<inputFile>" ).append( filename ).append( "</inputFile>\n" );
+    JobScheduleRequest jobScheduleRequest = new JobScheduleRequest();
+    jobScheduleRequest.setInputFile( filename );
 
     // Set the log level
     if ( meta.getLogLevel() != null ) {
-      sb.append( "<jobParameters>\n" );
-      sb.append( "<name>" ).append( "logLevel" ).append( "</name>\n" );
-      sb.append( "<type>" ).append( "string" ).append( "</type>\n" );
-      sb.append( "<stringValue>" ).append( meta.getLogLevel().getCode() ).append( "</stringValue>\n" );
-      sb.append( "</jobParameters>\n" );
+      JobScheduleParam jobScheduleParam = new JobScheduleParam();
+      jobScheduleParam.setName( "logLevel" );
+      jobScheduleParam.setType( "string" );
+      jobScheduleParam.setStringValue( new ArrayList<>( Arrays.asList( meta.getLogLevel().getCode() ) ) );
+      jobScheduleRequest.getJobParameters().add( jobScheduleParam );
     }
 
     // Set the clearing log param
-    sb.append( "<jobParameters>\n" );
-    sb.append( "<name>" ).append( "clearLog" ).append( "</name>\n" );
-    sb.append( "<type>" ).append( "string" ).append( "</type>\n" );
-    sb.append( "<stringValue>" ).append( meta.isClearingLog() ).append( "</stringValue>\n" );
-    sb.append( "</jobParameters>\n" );
+    JobScheduleParam jobScheduleParamClearingLog = new JobScheduleParam();
+    jobScheduleParamClearingLog.setName( "clearLog" );
+    jobScheduleParamClearingLog.setType( "string" );
+    jobScheduleParamClearingLog
+      .setStringValue( new ArrayList<>( Arrays.asList( String.valueOf( meta.isClearingLog() ) ) ) );
```

Comment:
`new ArrayList<>( Arrays.asList(`.... is this really necessary? See comments above about using `List`.